### ISSUE TITLE
Performance issue and copyright violation in `_sseclient.py` (for `db.Reference.listen()`)

### ISSUE THREAD

**Issue by daniel-ziegler** [firebase/firebase-admin-python#198]
### [REQUIRED] Step 2: Describe your environment

  * Operating System version: macOS 10.13.6
  * Firebase SDK version: 2.13.0
  * Firebase Product: database

#### Steps to reproduce:

1. Have some Firebase Realtime Database path containing a significant amount of data (>200KB, say).
2. Call `db.reference('/path').listen(callback)`.

As the first update, the server will send the entire contents of that path. On my machine this pegs a CPU core for minutes before the callback is finally called. The issue is the following code: https://github.com/firebase/firebase-admin-python/blob/abd39805f83f906296d5c99b6c4943754154d02c/firebase_admin/_sseclient.py#L93-L96
Since `self.resp_iterator` iterates through the stream one character at a time and `self._event_complete()` runs a regular expression over the entire buffer, this is an O(N^2) operation, which takes a long time when N is large.

**Separately, there is a copyright violation here.** I discovered that `_sseclient.py, despite having a Google copyright header, is largely copied from https://github.com/btubbs/sseclient/blob/master/sseclient.py.

#### Relevant Code:

Run the following code to reproduce:
```python
def listen_issue():
    path = '/sandbox/listen'

    print("Inserting data...")
    start = time.time()
    for i in range(10):
        long_data = ''.join(random.choice(string.ascii_uppercase) for _ in range(20000))
        db.reference(os.path.join(path, str(i))).set(long_data)
    end = time.time()
    print("...done (took %f seconds)" % (end - start))

    done_event = threading.Event()
    listener = db.reference(path).listen(lambda db_event: done_event.set())
    start = time.time()
    print("Waiting for first message from listen()...")
    done_event.wait()  # CPU usage is at 100% here
    end = time.time()
    print("...done (took %f seconds)" % (end - start))
    listener.close()
```

Example output on my machine:
```
Inserting data...
...done (took 1.678544 seconds)
Waiting for first message from listen()...
...done (took 109.180737 seconds)
```

**Comment by daniel-ziegler** [comment]
Any updates? I've worked around this with a healthy dose of monkey patching, but out of the box, the SDK is currently not usable for us.

**Comment by MeTaNoV** [comment]
@daniel-ziegler I will also use the stream feature introduced in last release. Apart from the performance issue you mentioned, is it functionally working?

**Comment by daniel-ziegler** [comment]
Yes, I believe it does work, although there's some complexity like handling both PUT and PATCH events.

**Comment by hiranya911** [comment]
Performance issue is simple enough to fix. As for the copyright issue, I believe some of this was borrowed from Pyrebase -- https://github.com/thisbejim/Pyrebase/blob/master/sseclient/sseclient.py; Although there's enough differences between the two, I'm not opposed to mentioning it in the file somewhere.

**Comment by hiranya911** [comment]
@daniel-ziegler can you try out the fix done in #221 and see if that solves your problem?
